tests/ha/check_after_reboot.pm

```diff
@@ -79,34 +81,33 @@ sub run {
 
     # Wait for resources to be started
     if (is_sles4sap) {
-        if (get_var('HA_CLUSTER_INIT') && check_var('CLUSTER_NAME', 'hana')) {
-            my $instance_id = get_required_var('INSTANCE_ID');
-            my $sid         = get_required_var('INSTANCE_SID');
-            my $sapadm      = lc($sid) . "adm";
-            my $node2       = choose_node(2);
-            if (check_var('AUTOMATED_REGISTER', 'false')) {
-                sleep bmwqemu::scale_timeout(300);
-                assert_script_run "su - $sapadm -c 'hdbnsutil -sr_register --name=NODE1 --remoteHost=$node2 --remoteInstance=$instance_id --replicationMode=sync --operationMode=logreplay'";
-                sleep 10;
-                assert_script_run "crm resource cleanup rsc_SAPHana_${sid}_HDB$instance_id", 300;
+        if (check_var('CLUSTER_NAME', 'hana') && check_var('AUTOMATED_REGISTER', 'false')) {
+            my $takeover_node = get_var('TAKEOVER_NODE');
+            if ($takeover_node ne get_hostname) {
+                check_cluster_state(proceed_on_failure => 1);
+                sles4sap->do_hana_takeover(node => $takeover_node, cluster => 1);
```

Comment:
I am not a big fan of calling a class method directly without instantiating an object, but it seems it can be done. My suggestion (if tidy allows and it and if it works) is to do the call as suggested in https://perldoc.perl.org/perlobj#Invoking-Class-Methods, i.e.:

`sles4sap::->do_hana_takeover(node => $takeover_node, cluster => 1);`

Reply:
OK, will do it.

Reply:
Just saw that it failed in travis. Guess we'll have to leave this as is.

Reply:
Syntax `'sles4sap'->do_hana_takeover(node => $takeover_node, cluster => 1);` works, pass the syntax tests and seems OK according to your doc.

Is it OK for you?

Reply:
Yes. That syntax also prevents the class to be evaluated as a method.